File: sheet 4 , J. Count Letters/main.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function countLetters(a){
    let char_count={};
    let cr;

    for (let index = 0; index < a.length; index++) {
        if (a[index] in char_count) {
            
            char_count[a[index]]+=1;
        }
        else{
            char_count[a[index]]=1;

        }
    }
    counter_sorted=Object.keys(char_count).sort()

    for (let index = 0; index < counter_sorted.length; index++) {
        console.log(`${counter_sorted[index]} : ${char_count[counter_sorted[index]]}`);
    }    

}


// Ask the user to enter a string
rl.question('', (userInput) => {
    countLetters(userInput)

    rl.close();
});

File: sheet 4 , J. Count Letters/main.py
def countLetters(a:str)->dict:
    char_count={}
    for cr in a:
        if cr in char_count:
            char_count[cr]+=1
        else:
            char_count[cr]=1
    return char_count

def showData(inpt:dict):
    count_sorted=sorted(inpt.keys())
    
    for cr in count_sorted:
        print(f"{cr} : {inpt[cr]}")
        
count=countLetters(input())
showData(count)

